services/provider/server/storageclusterpeer.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/red-hat-storage/ocs-operator/v4/services"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type storageClusterPeerManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClusterPeerManager(cl client.Client, namespace string) (*storageClusterPeerManager, error) {
+	return &storageClusterPeerManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func (s *storageClusterPeerManager) GetByStorageClusterUID(ctx context.Context, storageClusterUID types.UID) (*ocsv1.StorageClusterPeer, error) {
+	storageClusterPeerList := &ocsv1.StorageClusterPeerList{}
+	err := s.client.List(ctx, storageClusterPeerList, client.InNamespace(s.namespace))
+	if err != nil {
+		return nil, err
+	}
+	for i := range storageClusterPeerList.Items {
+		message, _, err := decodeTicket(storageClusterPeerList.Items[i].Spec.OnboardingToken)
+		if err != nil {
+			return nil, err
+		}
+
+		var ticketData services.OnboardingTicket
+		err = json.Unmarshal(message, &ticketData)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal onboarding ticket message. %v", err)
+		}
+
+		if ticketData.StorageCluster == storageClusterUID {
+			return &storageClusterPeerList.Items[i], nil
+		}
+	}
+	return nil, fmt.Errorf("StorageClusterPeer linked to StorageCluster with uid %q not found", storageClusterUID)
```

Comment:
Please read the UID from the status and not from the ticket.

Reply:
Status won't be available till this call is a success from the controller. It will cause a deadlock where API server waits for status and the controller populates the status only after sucess of this rpc call